From 7b6a0e5df4360ac2ae1d5f00d0650c9e9a0afb7c Mon Sep 17 00:00:00 2001
From: Rufus Hamade <rufus.hamade@imgtec.com>
Date: Tue, 22 Mar 2016 13:12:53 +0000
Subject: [PATCH 103/103] dri/pvr: Add extra information to
 PVRDRISyncPrimDelete

We need to decrement the object's reference count when we delete the
sync.  This requires that we pass some additional information to the
dri_support library.

Enhance PVRDRISyncPrimDelete to include these additional fields.
---
 src/mesa/drivers/dri/pvr/pvr_dri_support.h | 4 +++-
 src/mesa/drivers/dri/pvr/pvrsyncobj.c      | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/mesa/drivers/dri/pvr/pvr_dri_support.h b/src/mesa/drivers/dri/pvr/pvr_dri_support.h
index 2704df9..e0a3da2 100644
--- a/src/mesa/drivers/dri/pvr/pvr_dri_support.h
+++ b/src/mesa/drivers/dri/pvr/pvr_dri_support.h
@@ -293,7 +293,9 @@ void PVRDRIEGLImageGetAttribs(_EGLImage *psEGLImage, PVRDRIBufferAttribs *psAttr
 bool PVRDRISyncPrimAlloc(PVRDRIScreenImpl *psScreenImpl,
                          PVRSRV_CLIENT_SYNC_PRIM **psEGLSyncSyncPrim);
 
-void PVRDRISyncPrimDelete(PVRSRV_CLIENT_SYNC_PRIM *psEGLSyncSyncPrim);
+void PVRDRISyncPrimDelete(PVRDRIScreenImpl *psScreenImpl,
+                          PVRSRV_CLIENT_SYNC_PRIM *psEGLSyncSyncPrim,
+                          SyncCommand *psSyncCommand);
 
 bool PVRDRIInsertClientFence(pvrdri_api_type eAPI,
                              PVRDRIContextImpl *psContextImpl,
diff --git a/src/mesa/drivers/dri/pvr/pvrsyncobj.c b/src/mesa/drivers/dri/pvr/pvrsyncobj.c
index 87cbd0a..e964ea0 100644
--- a/src/mesa/drivers/dri/pvr/pvrsyncobj.c
+++ b/src/mesa/drivers/dri/pvr/pvrsyncobj.c
@@ -55,7 +55,9 @@ DeleteFence(__DRI2fence *psFence)
 {
 	if (psFence != NULL)
 	{
-		PVRDRISyncPrimDelete(psFence->psEGLFenceSyncPrim);
+		PVRDRISyncPrimDelete(psFence->psScreenImpl,
+		                     psFence->psEGLFenceSyncPrim,
+		                     psFence->psSyncCommand);
 		free(psFence);
 	}
 }
-- 
1.9.1

